### DISCUSSION TITLE
Only show Giscus comments in blog posts

### DISCUSSION THREAD

**Discussion by benlindsay** [quarto-dev/quarto-cli#3538] [Q&A] [answered]
I'm switching from Disqus to Giscus, and Giscus looks great, but I only want a Giscus comments section on blog posts, not on the blog post listing page or my About Me page. Is there a way to configure that without CSS hackers? Something like `posts-only: true` in the Giscus section of `_quarto.yml` or `comments: false` that can be added on a page by page basis?

**Comment by dragonstyle** [comment, marked as answer]
The simplest way is with a metadata file like this in your `posts` directory:

https://github.com/quarto-dev/quarto-web/tree/main/docs/blog/posts

which you can use to enable it for all the posts. You should also be able to set `comments: false` on a given page to disable comments specifically for that page.

HTH!

**Reply by benlindsay** [reply]
Oh cool! Didn't realize comments: false was an option or putting the giscus metadata in a subdirectory. Is there documentation to explain that that I missed?

**Reply by dragonstyle** [reply]
There isn't a single place that really shows the end to end process for this...

Commenting is a part of our HTML format, meaning that it can be specified per page (<https://quarto.org/docs/output-formats/html-basics.html#commenting>). This, in turn, means that you really can control it at the project level, at the `_metadata` level, or at the page level. It's not shown end to end in a single place, we're currently sort of relying on the principle explained above which I realize is a bit abstract. To get this out of our docs, you'd really have to read about and understand the way metadata works in Quarto (best example of that is here: <https://quarto.org/docs/projects/quarto-projects.html#shared-metadata>) and then note that `comments` is a format level option. Disabling is covered in the commenting docs, too.

Quarto is big enough that documentation is definitely a challenge and we're continuing to try to improve that!

**Reply by benlindsay** [reply]
Got it, thanks! Not sure how I missed the comments info on the HTML Basics page 😅
